lib/fndoc.nix

```diff
@@ -0,0 +1,15 @@
+{
+  docFn = {
+    description,
+    fn,
+    examples ? [],
+    example ? null
+  }:
+  let documentedFunction =
+    {
+      type = "documentedFunction";
+      __functor = self: x: fn x;
```

Comment:
If yes, what are the performance implications? Given that lib is used everywhere it might be problematic.